### PR TITLE
Add avahi explanation to Cast and Discovery docs

### PR DESCRIPTION
**Pull request by hmmbob**
## Proposed change

Adding explanation on how to set-up Avahi for Docker users that do not want to use the `--net:host` option. 

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [X] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: 
- Related issue: https://github.com/home-assistant/core/issues/34874

## Checklist

- [X] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [X] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documentation_standards.html

### CODE REVIEW

**Comment by hmmbob** [comment]
Lint fails on remarks that are in the code on purpose.

**Comment by klaasnicolaas** [comment]
@frenck Can you take a look at this PR?

File: source/_integrations/discovery.markdown [hunk 1]
```diff
@@ -107,12 +107,27 @@ Valid values for enable are:
 Home Assistant must be on the same network as the devices for UPnP discovery to work.
 If running Home Assistant in a [Docker container](/docs/installation/docker/) use switch `--net=host` to put it on the host's network.
 
+If for some reason you do not want to use `--net=host` you can also configure avahi-reflector on your host to make mDNS / UPnP work:
```
Comment (emontnemery) [inline source/_integrations/discovery.markdown]: The section should be renamed to `mDNS / UPnP`.
However, I think avahi-reflector only solves mDNS, not UPnP?

Comment (hmmbob) [reply source/_integrations/discovery.markdown]: I think you are right. Will fix.

Comment (hmmbob) [reply source/_integrations/discovery.markdown]: That triggers me.... shouldn't I update zeroconf docs as well?

**Comment by frenck** [comment]
We don’t support running Home Assistant this way, and should therefore not be documented in the official documentation.


Nevertheless, thanks for contributing!

**Comment by emontnemery** [comment]
@frenck just to clarify, what part is not supported?

**Comment by frenck** [comment]
running the Home Assistant Core off the host network.

**Pull request closed without merging**